### DISCUSSION TITLE
Foocus automate

### DISCUSSION THREAD

**Discussion by Buddy533** [lllyasviel/Fooocus#3266] [Q&A]
I need 2000 photos to process. I don't want to foocus each photo manually as it's time-consuming. Instead, I want to send 2000 photos and have the software process them one at a time. Is this possible?

**Comment by ILikeToasters** [comment]
You are not clear with what you are looking to do. You say process photos so not sure if you mean doing something with image prompts. Or do you just mean can you have it generate 2000 images in one go? If so you can edit the max number of images in the config.txt file in the Fooocus folder by adding the correct line. Or you can also right click on the generate button to have it generate forever. If you are talking about something like fixing faces automatically then the new enhance feature that should be added soon might be what you are looking for. You can already use that if you use Mashb1t's fork: https://github.com/mashb1t/Fooocus
<img width="258" alt="Screenshot 2024-07-11 150648" src="https://github.com/user-attachments/assets/3664b1ec-f478-4b7e-8add-1295229ace7b">

**Reply by Buddy533** [reply]
I am writing via deepl translator, I have 2000 photos and I need to expand each photo left and right, I have a specific promt to process these 2000 photos, and I am asking if it is possible to automatically select the 2000 files and let them be processed one by one according to my promt, or only manually drop each file into the "Inpaint or Outpaint" section and wait for it to be processed, and once processed I manually further send the next file, this process will be a long process

**Comment by mashb1t** [comment]
Feel free to use the Gradio Websockets API to send your requests to when you need image input to change, which you can find more information about in the footer.
You can find a working example in https://github.com/toutjavascript/FoooXus-Fooocus-Extender/blob/main/src/api.py and compare the parameters with your code.

Keep in mind that the limiting factor may be the Gradio max. queue size when sending thousands of requests.
